server/src-lib/Hasura/Eventing/EventTrigger.hs

```diff
@@ -393,10 +393,16 @@ logQErr err = do
   logger :: L.Logger L.Hasura <- asks getter
   L.unLogger logger $ EventInternalErr err
 
-getEventTriggerInfoFromEvent :: SchemaCache -> Event -> Maybe EventTriggerInfo
-getEventTriggerInfoFromEvent sc e = let table = eTable e
-                                        tableInfo = M.lookup table $ scTables sc
-                                    in M.lookup ( tmName $ eTrigger e) =<< (_tiEventTriggerInfoMap <$> tableInfo)
+getEventTriggerInfoFromEvent :: SchemaCache -> Event -> Either Text EventTriggerInfo
+getEventTriggerInfoFromEvent sc e = do
+  let table = eTable e
+      mTableInfo = M.lookup table $ scTables sc
+  tableInfo <- onNothing mTableInfo $ Left ("table '" <> table <<> "' not found")
```

Comment:
you can use `throwError` instead of explicitly constructing the value with `Left`.